### PR TITLE
WIP: new paraview layer structure

### PR DESCRIPTION
**Pull request by JonasLukasczyk**
Hi @julien-tierny, @pierre-guillou, and @CharlesGueunet,
this PR addresses issue #619.

In short, it removes all module folders from the paraview directory and instead provides a single directory called `paraview/xmls` that contains all `module.xml` files. I did not directly put these files into the paraview dir since there are two special folders: `singlePlugin` and `patch`. So the pv folder is now nice an clean. Note, there are some modules that did not have an xml file, but they still need to be registered at this step. I solved this issue by providing an "empty" xml file for these modules (it must at least contain the following string `<ServerManagerConfiguration></ServerManagerConfiguration>`). Similar to before where every module had to have a directory in the paraview folder, now every module has to have an xml file in the `paraview/xmls/` folder.

There are two TODOs before this can be merged in:
1. The create/rename/delete/clone scripts need to be updated
2. There was a `Compatibility.xml` file used to print some deprecation warnings that needs to be handled in a special way. I could do this, but I thought this might be a good time to remove these messages. So currently this xml file is removed.

Let me know what you think!

Best
Jonas

### CODE REVIEW

File: paraview/CMakeLists.txt [hunk 1]
```diff
@@ -57,3 +54,4 @@ install(
   DESTINATION
     "${TTK_INSTALL_PLUGIN_DIR}"
   )
+
```
Comment (vollia) [inline paraview/CMakeLists.txt]: avoid newline at the end of the file

**Review by CharlesGueunet — APPROVED** [review]

File: paraview/CMakeLists.txt (hunk 1 above)
Comment (JonasLukasczyk) [reply paraview/CMakeLists.txt]: done ;-)

**Comment by CharlesGueunet** [comment]
This new mechanism is way cleaner. I just find a bit strange the almost empty xml files for `ProgramBase`, ...

**Comment by JonasLukasczyk** [comment]
I agree with the empty files, they are necessary to still let the PV layer know which modules to build. Alternatively we could implement the following (more complicated logic):
For each directory in core/vtk register a pv_module and add the xml file if one exists. This way we do not need the empty files.

**Comment by CharlesGueunet** [comment]
let's keep the files like that, the simpler the better.

File: paraview/xmls/UserInterfaceBase.xml [hunk 2]
```diff
@@ -0,0 +1 @@
+<ServerManagerConfiguration></ServerManagerConfiguration>
```
Comment (julien-tierny) [inline paraview/xmls/UserInterfaceBase.xml]: I don't believe that file is even needed. 
(it's supposed to be used by standalone gui programs)

File: paraview/xmls/ProgramBase.xml [hunk 3]
```diff
@@ -0,0 +1 @@
+<ServerManagerConfiguration></ServerManagerConfiguration>
```
Comment (julien-tierny) [inline paraview/xmls/ProgramBase.xml]: same here (this vtk-layer class should only be used by standalone programs).
I think you can remove that.

**Review by julien-tierny — COMMENTED** [review]
could you remove the two entries I mentioned and update the script files as well?

**Comment by julien-tierny** [comment]
hi jonas,
thanks a lot for the PR.
ok for removing the compatibilty xml file.
if you could update the scripts and remove the two files I mentioned in my review, that'd be ready to ship.
thanks!

**Comment by JonasLukasczyk** [comment]
I deleted the two files and tests run fine. I also updated the clone and delete scripts, but the rename script is too complicated for me to quickly make changes. Can someone else have a look?

**Comment by julien-tierny** [comment]
@lhofmann 
to my understanding, the script `scripts/rename_plugins.py` was just a one-timer right?
(it had been used to migrate the name of the filters in the paraview plugin at some point)
I am not sure we need it anymore, do we?

**Comment by JonasLukasczyk** [comment]
If we do not need it I will delete it and then I guess this PR is ready to go.

**Comment by lhofmann** [comment]
Yes, `scripts/rename_plugins.py` is no longer needed.

**Comment by JonasLukasczyk** [comment]
Maybe we should let the developers know that if they are merging their branch with the dev then it is possible that they need to manually place the xml files of untracked modules in the xmls directory.

**Comment by julien-tierny** [comment]
alright, thanks for the confirmation @lhofmann and thanks @JonasLukasczyk for the cleanup!

**Pull request merged**